### PR TITLE
Create minio chart

### PR DESCRIPTION
**Pull request by lukkor**

### CODE REVIEW

File: minio/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+kind: Ingress
+metadata:
+  name: {{ template "minio.fullname" . }}
+  labels:
+    app: {{ template "minio.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  annotations:
+    {{- range $key, $value := .Values.ingress.annotations }}
```
Comment (gearnode) [inline minio/templates/ingress.yaml]: Why not use `to_yaml` here ?

File: minio/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
```
Comment (gearnode) [inline minio/templates/deployment.yaml]: Minio is a stateful app, we can use `statefulset` here.

File: minio/templates/network_policy.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+apiVersion: extensions/v1beta1
```
Comment (gearnode) [inline minio/templates/network_policy.yaml]: Wrap your network policy in condition. We don't alway need network policy, in minikube for example.

File: minio/templates/network_policy.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+metadata:
+  name: {{ template "minio.fullname" . }}
+  labels:
+    app: {{ template "minio.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  podSelector:
+    matchLabels:
```
Comment (gearnode) [inline minio/templates/network_policy.yaml]: Use the `fullname` to select, it's more clear

File: minio/templates/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ template "minio.fullname" . }}-pvc
```
Comment (gearnode) [inline minio/templates/pvc.yaml]: Any reason to suffix the name with `pvc` ?

File: minio/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    heritage: {{ .Release.Service }}
+spec:
+  type: ClusterIP
+  ports:
+    - port: 80
+      targetPort: 9000
+      protocol: TCP
+      name: http
+  selector:
+    app: {{ template "minio.name" . }}
```
Comment (gearnode) [inline minio/templates/service.yaml]: Use the `fullname` to select, it's more clear

File: minio/values.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Default values for minio.
+# This is a YAML-formatted file.
```
Comment (gearnode) [inline minio/values.yaml]: remove junk comment 😄

File: minio/values.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+persistence:
+  storageClassName: gp2
+  accessMode: ReadWriteOnce
+  size: 30Gi
+
+ingress:
+  enabled: true
+  # Used to create an Ingress record.
+  hosts:
+    - minio.beta.k8s.jobteaser.net
```
Comment (gearnode) [inline minio/values.yaml]: I am not ok with the default `host` in the ingress. This chart in not only for jobteaser.

File: minio/values.yaml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  tag: latest
+  pullPolicy: IfNotPresent
+
+persistence:
+  storageClassName: gp2
+  accessMode: ReadWriteOnce
+  size: 30Gi
+
+ingress:
+  enabled: true
```
Comment (gearnode) [inline minio/values.yaml]: I think the ingress should be false by default, no ?

**Comment by lukkor** [comment]
Minio chart already exists at https://github.com/kubernetes/charts/tree/master/stable/minio

**Pull request closed without merging**